controllers/spiaccesstokenbinding_controller.go

```diff
@@ -564,27 +677,141 @@ func (r *SPIAccessTokenBindingReconciler) syncSecret(ctx context.Context, sp ser
 	_, obj, err := r.syncer.Sync(ctx, nil, secret, secretDiffOpts)
 	if err != nil {
 		r.updateBindingStatusError(ctx, binding, api.SPIAccessTokenBindingErrorReasonTokenSync, err)
-		return api.TargetObjectRef{}, fmt.Errorf("failed to sync the secret with the token data: %w", err)
+		return nil, fmt.Errorf("failed to sync the secret with the token data: %w", err)
 	}
-	return toObjectRef(obj), nil
+	return obj.(*corev1.Secret), nil
+}
+
+func (r *SPIAccessTokenBindingReconciler) getOrCreateServiceAccount(ctx context.Context, binding *api.SPIAccessTokenBinding) (*corev1.ServiceAccount, error) {
```

Comment:
doc and test please

Reply:
This has 100% complete code coverage of the happy path by the integration tests and has a single error handler branch. Not sure what more you want to test.